src/drivers/uavcan/uavcan_main.hpp

```diff
@@ -209,7 +209,7 @@ class UavcanNode : public px4::ScheduledWorkItem, public ModuleParams
 	int		get_set_param(int nodeid, const char *name, uavcan::protocol::param::GetSet::Request &req);
 	void 		update_params();
 
-	void		set_setget_response(uavcan::protocol::param::GetSet::Response *resp) { _setget_response = resp; }
+	void		set_setget_response(uavcan::protocol::param::GetSet::Response *resp) { memcpy(_setget_response, resp, sizeof(*resp)); }
```

Comment:
What was wrong with this?

Reply:
if the argument of `set_setget_response ()`, `resp`, is the address of a local variable, then the current implementation would dangling object, and memory leak may happen in that case.

Reply:
@howjmay each method that calls `set_setget_response`, also calls `free_setget_response`, so the local variable isn't accessed outside of the method.

Reply:
reverted the change!